spec/spec_helper.rb

```diff
@@ -14,6 +14,10 @@
 require 'vcr'
 require 'cgi'
 
+# Fail tests that try to include stuff in `main`
+require_relative 'support/test_contamination'
+Spec::Support::TestContamination.setup(self)
```

Comment:
I don't know how much of a fan I am of the parameterized setup passing `self`. It seems ok, but also seems overly engineered in some way (since it's always going to be on `Object`). Is there a case we'd use this beyond global namespace polluting?

If not, maybe just keep the patched_include in `Spec::Support:TestContamination` and call it with `Object.prepend(Spec::Support....` in spec helper? ¯\\\_(ツ)_/¯ Random thoughts on that, but if others think this is perfect, that's cool too.